packages/passwordless-auth-construct/src/sign-up/construct.ts

```diff
@@ -0,0 +1,103 @@
+import * as path from 'path';
+import { fileURLToPath } from 'url';
+import { Architecture, Runtime } from 'aws-cdk-lib/aws-lambda';
+import * as apigateway from 'aws-cdk-lib/aws-apigateway';
+import { IUserPool } from 'aws-cdk-lib/aws-cognito';
+import { Effect, Policy, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { CfnOutput } from 'aws-cdk-lib';
+
+const filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(filename);
+
+/**
+ * Amplify SignUp Passwordless construct
+ */
+export class AmplifySignUpPasswordless extends Construct {
+  /**
+   * Create a new SignUp Passwordless construct
+   * verifyAuthChallengeResponse function is needed in order to attach policy
+   * userPool is needed to identity the resource and create the policy for verifyAuthChallengeResponse
+   */
+  constructor(
+    scope: Construct,
+    id: string,
+    verifyAuthChallengeResponse: NodejsFunction,
+    userPool: IUserPool
+  ) {
+    super(scope, id);
+
+    // default memory allocation for lambda functions
+    const defaultMemorySize = 128;
+
+    const commonOptions: NodejsFunctionProps = {
+      entry: path.join(dirname, '.', 'create_user_handler.js'),
+      runtime: Runtime.NODEJS_20_X,
+      architecture: Architecture.ARM_64,
+      bundling: {
+        format: OutputFormat.ESM,
+      },
+    };
+
+    const createUserFunction = new NodejsFunction(scope, `CreateUser${id}`, {
+      handler: 'createUser',
+      ...commonOptions,
+      memorySize: defaultMemorySize,
+    });
+
+    const api = new apigateway.RestApi(this, `CreateUserApi${id}`, {
+      restApiName: 'Create User service',
+      description: 'This service creates users.',
+      defaultCorsPreflightOptions: {
+        allowOrigins: apigateway.Cors.ALL_ORIGINS,
+      },
+    });
+
+    const putCreateUserIntegration = new apigateway.LambdaIntegration(
+      createUserFunction,
+      {
+        requestTemplates: { 'application/json': '{ "statusCode": "200" }' },
+      }
+    );
+
+    api.root.addMethod('PUT', putCreateUserIntegration);
+
+    new CfnOutput(this, 'apiUrl', { value: api.url });
+
+    // According to cdk docs there always is a default role https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda-readme.html#execution-role
+    // eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
+    createUserFunction.role!.attachInlinePolicy(
+      new Policy(this, `CreateUserPolicy${id}`, {
+        statements: [
+          new PolicyStatement({
+            actions: ['cognito-idp:AdminCreateUser'],
+            resources: [userPool.userPoolArn],
+            effect: Effect.ALLOW,
+          }),
+        ],
+      })
+    );
+
+    // According to cdk docs there always is a default role https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda-readme.html#execution-role
+    // eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
+    verifyAuthChallengeResponse.role!.attachInlinePolicy(
```

Comment:
If we only need the function execution role, that should be what's passed into the ctor rather than the whole function